### PR TITLE
codeintel: Do not perform data-retention scans on stale repositories

### PR DESCRIPTION
**Pull request by efritz**
Same effort as #25571. There is ~an obvious bug~ obviously a bug that causes valid records to be deleted while the repository's commit graph is stale. I noticed this on the codeintel-qa pipeline, which always starts with a fresh instance. We were not seeing this on local instances which were already running because the test timing happened to miss the data retention janitor. Ensuring they run concurrently shows a lot of records marked as expired that shouldn't be.

This PR changes the data retention process to only work on repositories with updated commit graphs. This is a simplifying condition, and the only change in behavior here is the frequency that records are removed for repositories _that are always stale_. This is another issue entirely.

### CODE REVIEW

File: enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go [hunk 1]
```diff
@@ -48,13 +48,13 @@ func NewUploadExpirer(
 	interval time.Duration,
 	metrics *metrics,
 ) goroutine.BackgroundRoutine {
-	return goroutine.NewPeriodicGoroutine(context.Background(), interval, &uploadExpirer{
+	return goroutine.NewPeriodicGoroutine(context.Background() /*interval*/, time.Second, &uploadExpirer{
```
Comment (eseliger) [inline enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go:51]: Intentional?

File: enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go [hunk 2]
```diff
@@ -48,13 +48,13 @@ func NewUploadExpirer(
...
 		dbStore:                dbStore,
 		gitserverClient:        gitserverClient,
 		metrics:                metrics,
-		repositoryProcessDelay: repositoryProcessDelay,
+		repositoryProcessDelay: time.Second, // repositoryProcessDelay,
 		repositoryBatchSize:    repositoryBatchSize,
-		uploadProcessDelay:     uploadProcessDelay,
+		uploadProcessDelay:     time.Second, // uploadProcessDelay,
```
Comment (eseliger) [inline enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go:57]: Intentional?

File: enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go:51]: NO

File: enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go (hunk 2 above)
Comment (efritz) [reply enterprise/cmd/worker/internal/codeintel/janitor/upload_expirer.go:57]: NO

**Review by eseliger — APPROVED** [review]

**Pull request merged**